include/deal.II/simplex/grid_generator.h

```diff
@@ -27,221 +27,225 @@
 
 DEAL_II_NAMESPACE_OPEN
 
-namespace Simplex
+namespace GridGenerator
 {
   /**
-   * This namespace provides a collection of functions to generate simplex
-   * triangulations for some basic geometries.
+   * Create a coordinate-parallel brick from the two diagonally opposite
+   * corner points @p p1 and @p p2. The number of vertices in coordinate
+   * direction @p i is given by <tt>repetitions[i]+1</tt>.
    *
-   *  @ingroup simplex
+   * @note This function connects internally 4/8 vertices to quadrilateral/
+   *   hexahedral cells and subdivides these into 2/5 triangular/
+   *   tetrahedral cells.
+   *
+   * @note Currently, this function only works for `dim==spacedim`.
+   *
+   * @ingroup simplex
    */
-  namespace GridGenerator
+  template <int dim, int spacedim>
+  void
+  subdivided_hyper_rectangle_with_simplices(
+    Triangulation<dim, spacedim> &   tria,
+    const std::vector<unsigned int> &repetitions,
+    const Point<dim> &               p1,
+    const Point<dim> &               p2,
+    const bool                       colorize = false)
   {
```

Comment:
What's the reason to have the implementation of this function in the .h file, by the way?